sdk/common/balances.ts

```diff
@@ -0,0 +1,47 @@
+import { ContractsMap, CurrencyKey } from '@synthetixio/contracts-interface';
+import { SynthBalancesMap } from '@synthetixio/queries';
+import { wei } from '@synthetixio/wei';
+import { ethers } from 'ethers';
+import { orderBy } from 'lodash';
+
+import { notNill } from 'queries/synths/utils';
+import { zeroBN } from 'utils/formatters/number';
+
+type SynthBalancesTuple = [string[], ethers.BigNumber[], ethers.BigNumber[]];
+
+export const getSynthBalances = async (walletAddress: string, contracts: ContractsMap) => {
```

Comment:
I was thinking this could also be a class / service similar to exchange / futures etc, as it will also be api's we'd like to expose to anyone using the sdk rather than internal methods that need to be passed contract instances.

Reply:
We could follow another pattern from Yearn for this, where they pass the root sdk class through to each service: https://github.com/yearn/yearn-sdk/blob/main/src/common.ts#L24

Reply:
Each class then extends the service interface: https://github.com/yearn/yearn-sdk/blob/main/src/interfaces/token.ts#L29
With `this` provided on init: https://github.com/yearn/yearn-sdk/blob/main/src/yearn.ts#L117
allowing each service to access any other part of the sdk: https://github.com/yearn/yearn-sdk/blob/main/src/interfaces/token.ts#L39